### ISSUE TITLE
pullEvent filters ignored

### ISSUE THREAD

**Issue by apemanzilla** [Lignumm/Radium#7]
Calling os.pullEvent with a filter will return upon the first event received, whether the filter matches or not. When a coroutine yields, it returns the pullEvent filter, which can be captured like so:

```lua
local function test()
  os.pullEvent("test_filter")
end

local c = coroutine.create(test)
local ok, filter = coroutine.resume(c)
print(filter)
```

In this example, "test_filter" should be printed. prcomgr.distribute should be modified to respect event filters, only resuming coroutines if the appropriate event is caught.

**Comment by Lignumm** [comment]
Woops, turns out I completely forgot to implement filters at all. I'm surprised so many programs worked so far though :P.

**Comment by Lignumm** [comment]
Solved as of 3ac741960d6372d85984210b842c745bd0404ad5.
